accounts-db/src/accounts_db.rs

```diff
@@ -5079,6 +5113,44 @@ impl AccountsDb {
             }
         };
 
+        let mut limit = 2;
+        let mut added = 0;
+        if shrink_slots.len() >= limit {
+            limit = shrink_slots.len() + 1;
+        }
+        let mut num_found = 0;
+        if shrink_slots.len() < limit {
```

Comment:
This condition seems to be always true: shrink_slots().len is 0 or 1, limit is 2, shrink_slots.len() is 2, 3, ... limit is 3, 4, ... shrink_slots.len() + 1.

I'm trying to understand the purpose of this entire block. We add something to the shrink_slots only if the something's capacity is the same as storage capacity for the corresponding slot, but only one such addition is allowed (or two if we start with an empty list of shrink_slots). Could you, maybe, in a few words help me to figure out what we're accomplishing here in the context of the whole shrinking algorithm?